src/components/ComponentListView/ComponentListViewPage.tsx

```diff
@@ -0,0 +1,58 @@
+import * as React from 'react';
+import { Link } from 'react-router-dom';
+import {
+  DataList,
+  DataListCell,
+  DataListItem,
+  DataListItemCells,
+  DataListItemRow,
+} from '@patternfly/react-core';
+import { useK8sWatchResource } from '../../dynamic-plugin-sdk';
+import { useActiveNamespace } from '../../hooks/useActiveNamespace';
+import { ComponentGroupVersionKind } from '../../models';
+import { StatusBox } from '../../shared/components/status-box/StatusBox';
+import { ComponentKind } from '../../types';
+import { ComponentListItem } from './ComponentListItem';
+
+type ComponentListViewPageProps = {
+  application: string;
+};
+
+export const ComponentListViewPage: React.FC<ComponentListViewPageProps> = ({ application }) => {
+  const namespace = useActiveNamespace();
+  const [allComponents, componentsLoaded] = useK8sWatchResource<ComponentKind[]>({
+    groupVersionKind: ComponentGroupVersionKind,
+    namespace,
+    isList: true,
+  });
+  const loaded = namespace && componentsLoaded;
```

Comment:
We need to have this check since the active namespace needs to be fetched and it's set to null on first render when making the fetch call for components.